test/common_utils.py

```diff
@@ -662,9 +662,13 @@ def random_symmetric_psd_matrix(l):
     return A.mm(A.transpose(0, 1))
 
 
-def random_symmetric_pd_matrix(l, eps=1e-5):
-    A = torch.randn(l, l)
-    return A.mm(A.transpose(0, 1)) + torch.eye(l) * eps
+def random_symmetric_pd_matrix(l, *batches):
+    if len(batches) == 0:
+        A = torch.randn(l, l)
+        return A.mm(A.transpose(0, 1)) + torch.eye(l) * 1e-5
+    else:
+        A = torch.randn(*(batches + (l, l)))
```

Comment:
Shouldn't this work even when len(batches) = 0? I don't think len(batches)= 0 needs to be special cased

Reply:
Yes, you are right. Sorry about that.